src/draggable/DraggingWrapper.tsx

```diff
@@ -0,0 +1,100 @@
+// we could replace dnd with this https://docs.dndkit.com
+import Grid2 from '@mui/material/Unstable_Grid2/Grid2';
+
+import { DndProvider } from 'react-dnd';
+import { HTML5Backend } from 'react-dnd-html5-backend';
+
+import DraggableElement, {
+  DraggableElementProps,
+  DroppedFile,
+} from './DraggableElement';
+import InBetween, { InBetweenProps } from './InBetween';
+
+export type DraggingWrapperProps<T> = {
+  renderComponent: DraggableElementProps<T>['renderComponent'];
+
+  rows?: T[];
+
+  /** wrapper id */
+  id?: string;
+  getRowId?: (el: T) => string;
+
+  /** show checkbox */
+  // showCheckbox?: boolean;
+  /** show drag anchor */
+  isMovable?: DraggableElementProps<T>['isMovable'];
+  /** handler on drop in a row */
+  onDropInRow?: DraggableElementProps<T>['onDrop'];
+  /** enable to drag in between rows */
+  enableMoveInBetween?: boolean;
+  /** handler on drop in between rows */
+  onDropBetweenRow?: InBetweenProps<T>['onDrop'];
+
+  allowFiles?: DraggableElementProps<T>['allowFiles'];
+
+  canDrop?: DraggableElementProps<T>['canDrop'];
+
+  xs?: number;
+};
+
+const DraggingWrapper = <T extends object>({
+  id,
+  rows = [],
+  getRowId,
+  onDropInRow: onDropInRowFn,
+  onDropBetweenRow: onDropBetweenRowFn,
+  renderComponent,
+  isMovable,
+  enableMoveInBetween = true,
+  allowFiles = true,
+  canDrop,
+  xs = 12,
```

Comment:
How are you planning to use this property ? If it is to create more columns depending on the breakpoint, directly set the number based on the breakpoint.